src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -821,147 +828,145 @@ public unsafe void CopyTo(Array array, int index)
-                            Vector128<byte> vector = Vector128.Create(bits).AsByte();
-                            vector = AdvSimd.Arm64.ZipLow(vector, vector);
-                            vector = AdvSimd.Arm64.ZipLow(vector, vector);
-
-                            Vector128<byte> shuffledLower = AdvSimd.Arm64.ZipLow(vector, vector);
-                            Vector128<byte> extractedLower = AdvSimd.And(shuffledLower, bitMask128);
-                            Vector128<byte> normalizedLower = AdvSimd.Min(extractedLower, ones);
-
-                            Vector128<byte> shuffledHigher = AdvSimd.Arm64.ZipHigh(vector, vector);
-                            Vector128<byte> extractedHigher = AdvSimd.And(shuffledHigher, bitMask128);
-                            Vector128<byte> normalizedHigher = AdvSimd.Min(extractedHigher, ones);
-
-                            AdvSimd.Arm64.StorePair((byte*)destination + i, normalizedLower, normalizedHigher);
-                        }
+                        Vector128<int> bits = Vector128.CreateScalarUnsafe(Unsafe.AddByteOffset(ref thisRef, i >> BitShiftPerByte));
+
+
+                        Vector128<byte> shuffledLower = Vector128.Shuffle(bits.AsByte(), lowerShuffleMask);
+                        Vector128<byte> extractedLower = shuffledLower & bitMask;
+
+                        // The extracted bits can be anywhere between 0 and 255, so we normalise the value to either 0 or 1
+                        // to ensure compatibility with "C# bool" (0 for false, 1 for true, rest undefined)
+                        Vector128<byte> normalizedLower = Vector128.Min(extractedLower, ones);
+                        normalizedLower.StoreUnsafe(ref Unsafe.As<bool, byte>(ref boolRef), i);
+                        i += (uint)Vector128<byte>.Count;
+
+
+                        Vector128<byte> shuffledUpper = Vector128.Shuffle(bits.AsByte(), upperShuffleMask);
+                        Vector128<byte> extractedUpper = shuffledUpper & bitMask;
+
+                        // The extracted bits can be anywhere between 0 and 255, so we normalise the value to either 0 or 1
+                        // to ensure compatibility with "C# bool" (0 for false, 1 for true, rest undefined)
+                        Vector128<byte> normalizedUpper = Vector128.Min(extractedUpper, ones);
+                        normalizedUpper.StoreUnsafe(ref Unsafe.As<bool, byte>(ref boolRef), i);
+                        i += (uint)Vector128<byte>.Count;
                     }
                 }
 
             LessThan32:
-                for (; i < (uint)m_length; i++)
+                for (; i < (uint)thisLength; i++)
                 {
                     int elementIndex = Div32Rem((int)i, out int extraBits);
-                    boolArray[(uint)index + i] = ((m_array[elementIndex] >> extraBits) & 0x00000001) != 0;
+                    Unsafe.Add(ref boolRef, i) = ((Unsafe.Add(ref thisRef, elementIndex) >> extraBits) & 0x00000001) != 0;
```

Comment:
Can you revert this part back to original safe code?

Reply:
Sure. Later in the next commit.